File: net/JetBrains.Serialization/src/Dataholders/ExtensionInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;

namespace JetBrains.Serialization;

[JsonObject(MemberSerialization.OptIn)]
public class ExtensionInfo : IEncodableInfo
{
  [JsonProperty("Key")] private TextualInfo _key;
  [JsonProperty("Critical")] private bool _critical;
  [JsonProperty("Value")] private TextualInfo _value;

  [JsonConstructor]
  public ExtensionInfo(TextualInfo key, bool critical, TextualInfo value)
  {
    _key = key;
    _critical = critical;
    _value = value;
  }

  private DerSequence ToDLSequence() =>
    new List<Asn1Encodable?>
      { _key.ToPrimitive(), _critical ? DerBoolean.True : null, _value.ToPrimitive() }.ToDerSequence();

  public Asn1Encodable ToPrimitive() => ToDLSequence();
}

File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/dataholders/ExtensionInfo.kt
package com.jetbrains.signatureverifier.serialization.dataholders

import com.jetbrains.signatureverifier.serialization.toDLSequence
import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.ASN1Boolean
import org.bouncycastle.asn1.ASN1Primitive
import org.bouncycastle.asn1.DLSequence
@Serializable
data class ExtensionInfo(
  val key: TextualInfo,
  val critical: Boolean,
  val value: TextualInfo
) : EncodableInfo {
  private fun toDLSequence(): DLSequence = listOf(
    key.toPrimitive(),
    (if (critical) ASN1Boolean.TRUE else null),
    value.toPrimitive()
  ).toDLSequence()

  override fun toPrimitive(): ASN1Primitive = toDLSequence().toASN1Primitive()
}